distributed/worker.py

```diff
@@ -3423,11 +3434,10 @@ def done_event():
                 )
                 self.update_who_has(who_has)
 
-    @log_errors
-    def _readd_busy_worker(self, worker: str) -> None:
-        self.busy_workers.remove(worker)
-        self.handle_stimulus(
-            GatherDepDoneEvent(stimulus_id=f"readd-busy-worker-{time()}")
+    async def retry_busy_worker_later(self, worker: str) -> StateMachineEvent | None:
+        await asyncio.sleep(0.15)
+        return RetryBusyWorkerEvent(
+            worker=worker, stimulus_id=f"retry-busy-worker-{time()}"
```

Comment:
For the sake of keeping unit tests simple, I think that in the WorkerBase class this should look like
```python
class WorkerBase(abc.ABC):
    @abc.abstractmethod
    async def gather_dep(...) -> StateMachineEvent | None: ...

    @abc.abstractmethod
    async def compute(...) -> StateMachineEvent | None: ...

    async def retry_busy_worker_later(self, worker: str) -> StateMachineEvent | None:
        return RetryBusyWorkerEvent(
            worker=worker, stimulus_id=f"retry-busy-worker-{time()}"
        )
"""